File: algorithms/python/binary_search.py
def binary_search(mas,item):
	start = 0
	end = len(mas)-1
	while(start < end):
		mid = int((start + end) / 2)
		quess = mas[mid]
		if(quess == item):
			return mid
		if(quess < item):
			start = mid
		else:
			end = mid
	return None

def main():
	l = [0,1,2,3,4,5]
	a = 2
	n = -1
	print(l)
	print("BS(2):")
	print(binary_search(l,a))

	print("BS(-1):")
	print(binary_search(l,n))

if __name__ == '__main__':
	main()

File: algorithms/c/binary_search.c
#include <stdio.h>

int binary_search(int arr[], int size, int item){
	int start = 0;
	int end = size;

	while(start < end){
		int midle = (end - start) / 2;
		int quess = arr[midle];

		if(quess == item){
			return midle;
		}
		if(quess < item){
			start = midle;
		}
		else{
			end = midle;
		}

	}
}

void test(){
	int mas[5] = {0,3,27,50,122};
	printf("0-3-27-50-122\nindex of three: %d\n", binary_search(mas, 5, 3));
}

int main(int argc, char const *argv[]){
	test();
	return 0;
}